File: 0789-kth-largest-element-in-a-stream/0789-kth-largest-element-in-a-stream.java
class KthLargest {
    PriorityQueue<Integer> queue;
    int k;
    public KthLargest(int k, int[] nums) {
        this.k = k;
        queue = new PriorityQueue<>();
        for (int num : nums) {
            queue.add(num);
            if (queue.size() > k) {
                queue.remove();
            }
        }
    }
    
    public int add(int val) {
        queue.add(val);
        if (queue.size() > this.k) {
            queue.remove();
        }
        
        return queue.peek();
    }
}

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest obj = new KthLargest(k, nums);
 * int param_1 = obj.add(val);
 */

File: 0789-kth-largest-element-in-a-stream/0789-kth-largest-element-in-a-stream.py
class KthLargest:

    def __init__(self, k: int, nums: List[int]):
        self.queue = []
        self.k = k
        
        for num in nums:
            heapq.heappush(self.queue, num)

            if len(self.queue) > k:
                heapq.heappop(self.queue)

    def add(self, val: int) -> int:
        heapq.heappush(self.queue, val)

        if len(self.queue) > self.k:
            heapq.heappop(self.queue)
        
        return self.queue[0]


# Your KthLargest object will be instantiated and called as such:
# obj = KthLargest(k, nums)
# param_1 = obj.add(val)